x-pack/plugins/observability_solution/infra/server/lib/helpers/get_infra_metrics_client.ts

```diff
@@ -29,19 +33,40 @@ export async function getInfraMetricsClient({
 }) {
   const { framework } = libs;
   const infraContext = await context.infra;
+  const { uiSettings } = await context.core;
+
+  const excludedDataTiers = await uiSettings.client.get<DataTier[]>(searchExcludedDataTiers);
   const metricsIndices = await infraContext.getMetricsIndices();
 
   return {
     search<TDocument, TParams extends RequiredParams>(
       searchParams: TParams
     ): Promise<InferSearchResponseOf<TDocument, TParams>> {
+      const searchFilter = searchParams.body.query?.bool?.filter ?? [];
+
+      const excludedQuery = excludedDataTiers.length ? excludeTiersQuery(excludedDataTiers) : [];
+
+      // This flattens arrays by one level, and non-array values can be added as well, so it all
+      // results in a nice [QueryDsl, QueryDsl, ...] array.
+      const filter = ([] as QueryDslQueryContainer[]).concat(searchFilter, excludedQuery);
+
       return framework.callWithRequest(
         context,
         'search',
         {
           ...searchParams,
           ignore_unavailable: true,
           index: metricsIndices,
+          body: {
+            ...searchParams.body,
+            query: {
+              ...searchParams.body.query,
+              bool: {
+                ...searchParams.body.query?.bool,
+                filter,
```

Comment:
the  `searchParams.body.query?.bool` object is spread into the new `body.query.bool` object, so if there are any pre-existing `bool` params, it all gets placed into the same object. `filter` is a prop within the `bool` interface, so that gets overridden by the concatenated value of any `filter` present in the `searchParams.body.query?.bool` object with the `excludedQuery` derived from the ui settings.

What you see is basically me having carefully verified the interface typing as to what the resulting query shape should look like, while trying to preserve any existing options that come into the request parameters.

I could tweak this a bit further to maybe even remove the query object spreading if there are no filters to be injected.

Reply:
oh, nvm. just saw this `const searchFilter = searchParams.body.query?.bool?.filter ?? [];`